packages/swingset-theme-hashicorp/src/components/code-block/code-block.tsx

```diff
@@ -0,0 +1,43 @@
+'use client' //Library calls hooks internally
+import { Highlight, themes } from 'prism-react-renderer'
+import { parseCode, parseLanguage } from './helpers'
+import { Language, MDXPreClass } from '@/types'
+import { CopyButton } from './copy-button'
+
+type CodeBlockProps = {
+  filePath: `${string}.${Language}`
+  children: React.Component
+  className: MDXPreClass
+}
+
+function CodeBlock(props: CodeBlockProps) {
+  const { children } = props
+  const code = parseCode(children)
+  const { className } = children.props as typeof children.props & {
+    className: MDXPreClass
+  }
+  const language = parseLanguage(className || 'language-jsx')
```

Comment:
Validates that the language is supported, else defaults to jsx.
As mentioned in the description only ts,js,jsx and tsx are supported which works for now but if we want to be flexible, we should try to adopt the Hashicorp `CodeBlock`